src/MongoDB.Driver.Core/Core/WireProtocol/Messages/Encoders/BinaryEncoders/MessageBinaryEncoderBase.cs

```diff
@@ -68,6 +68,20 @@ protected UTF8Encoding Encoding
             }
         }
 
+        /// <summary>
+        /// Gets a flag whether encryption has been configured.
+        /// </summary>
+        /// <value>
+        /// The flag whether encryption is configured or not.
+        /// </value>
+        protected bool IsEncryptionConfigured
```

Comment:
This should not be needed.

Reply:
Resolving this comment because we decided to keep this quick and dirty workaround to the fact that MaxDocumentSize was configured TWO different ways.